nnvm/python/nnvm/frontend/keras.py

```diff
@@ -530,17 +570,21 @@ def from_keras(model):
                 # The one exception is InputLayer.  Changing input variable names after conversion
                 # would confuse users, so we should keep them as far as possible.  Fortunately,
                 # they are named uniquely to input_1, input_2, input_3 ... by default.
-                for pred_idx, pred in zip(node.node_indices, node.inbound_layers):
+                zip_node = zip(node.node_indices, node.tensor_indices, node.inbound_layers)
+                for pred_idx, t_idx, pred in zip_node:
                     if isinstance(pred, keras.engine.InputLayer):
                         sym = symtab.get_var(pred.name, must_contain=True)
                     else:
-                        sym = symtab.get_var(pred.name + ':' + str(pred_idx), must_contain=True)
+                        sym_name = pred.name + ':' + str(pred_idx) + ':' + str(t_idx)
+                        sym = symtab.get_var(sym_name, must_contain=True)
                     insym.append(sym)
 
                 if len(insym) == 1:
                     insym = insym[0]
                 keras_op_to_nnvm(insym, keras_layer, keras_layer.name + ':' + str(my_idx), symtab)
 
-    outsym = [symtab.get_var(layer.name + ':0') for layer in model._output_layers]
+    outsym = [symtab.get_var(oc[0].name + ":" + str(oc[1]) + ":" + str(oc[2]))
+              for oc in model._output_coordinates]
```

Comment:
`for layer, n_idx, t_idx in model._output_coordinates` to make it clear what _output_coordinates() returns.